packages/core/src/util.ts

```diff
@@ -232,3 +232,23 @@ export function deepCompare<T>(a: T, b: T): boolean {
 
   return true;
 }
+
+export const createPromise = <T>(
```

Comment:
Maybe a more descriptive name like `createTimerPromise` or `createTimeoutPromise` ?

Reply:
it used to be `createTimeoutPromise` but then I realized I made the timeout optional.